src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -147,10 +147,61 @@ def _snap_to_rail(string: str, rails: Set[str]) -> Optional[str]:
     """
 
     processed_string = string.strip()
-    if processed_string not in rails:
+    rails_list = list(rails)
+    rail = _detect_substring(processed_string, rails_list[0], rails_list[1])
+    if not rail:
         logger.warning(
-            f"LLM output cannot be snapped to rails {list(rails)}, returning None. "
+            f"LLM output cannot be snapped to rails {list(rails)}, returning NOT_PARSABLE. "
             f'Output: "{string}"'
         )
-        return None
-    return processed_string
+        return "NOT_PARSABLE"
+    return rail
+
+def _detect_substring(Z, A, B):
```

Comment:
```suggestion
def _extract_rail(output, rails):
```
detect implies a boolean response, extract implies parsing